#**Crescimento Populacional Mundial (EDA)🌎👪**

##**Análise dos dados**

In [166]:
import pandas as pd
import plotly.express as px

In [167]:
dt_cresc_pop = pd.read_csv('/content/drive/MyDrive/DFs /dataset_cresPopulacional.csv')

In [168]:
dt_cresc_pop.shape
#101 linhas, 3 colunas.

(101, 3)

In [169]:
dt_cresc_pop.info

<bound method DataFrame.info of                        Country/Region Ø Growth/year Growth 2013-2022
0                   Equatorial Guinea         5.08%           55.60%
1                               Niger         3.95%           41.63%
2    Democratic Republic of the Congo         3.68%           38.75%
3                              Gambia         3.60%           37.80%
4                              Angola         3.55%           36.80%
..                                ...           ...              ...
96                           Bulgaria        -1.20%          -11.01%
97                     Saint Martin *        -1.51%          -12.80%
98                            Ukraine        -1.72%          -16.46%
99                   American Samoa *        -2.24%          -20.54%
100                  Marshall Islands        -2.98%          -26.99%

[101 rows x 3 columns]>

In [170]:
dt_cresc_pop.head()

Country/Region Ø Growth/year Growth 2013-2022
0                 Equatorial Guinea         5.08%           55.60%
1                             Niger         3.95%           41.63%
2  Democratic Republic of the Congo         3.68%           38.75%
3                            Gambia         3.60%           37.80%
4                            Angola         3.55%           36.80%

In [171]:
dt_cresc_pop.isnull().sum()

Country/Region      0
Ø Growth/year       0
Growth 2013-2022    0
dtype: int64

In [172]:
dt_cresc_pop.columns

Index(['Country/Region', 'Ø Growth/year', 'Growth 2013-2022'], dtype='object')

#**Taxa de Crescimento Populacional Anual**

In [173]:
#Realizando a conversão da coluna Growth/year para float.
dt_cresc_pop['Ø Growth/year'] = dt_cresc_pop['Ø Growth/year'].str.replace('%', '').astype(float)

##**Top 10 Países com a maior taxa de Crescimento Populacional Anual**

In [174]:
dt_cresc_pop.groupby('Country/Region')[['Ø Growth/year']].max().reset_index().sort_values('Ø Growth/year', ascending=False)

Country/Region  Ø Growth/year
30                 Equatorial Guinea           5.08
62                             Niger           3.95
25  Democratic Republic of the Congo           3.68
35                            Gambia           3.60
4                             Angola           3.55
..                               ...            ...
15                          Bulgaria          -1.20
76                    Saint Martin *          -1.51
93                           Ukraine          -1.72
3                   American Samoa *          -2.24
55                  Marshall Islands          -2.98

[101 rows x 2 columns]

In [175]:
top10_growth_year = dt_cresc_pop.groupby('Country/Region')[['Ø Growth/year']].max().reset_index().sort_values('Ø Growth/year', ascending=False)
top10_growth_year = top10_growth_year.head(10)

In [176]:
#Plotagem Gráfica
top10_graphic = px.bar(top10_growth_year,
                x='Country/Region',
                y='Ø Growth/year',
                title='Os 10 Países com a maior taxa de Crescimento Populacional Anual',
                color='Ø Growth/year',
                color_continuous_scale='blues'
                )

top10_graphic.update_layout(xaxis_title='País/Região',
                            yaxis_title='Taxa de Crescimento Anual'
                            )
top10_graphic.show()

Podemos observar com a plotagem gráfica que o País com a maior taxa de crescimento populacional anual é a **Guinea Equatorial com uma taxa de 5.08% ao ano.**

##**Top 10 Países com a menor taxa de Crescimento Populacional Anual**

In [177]:
top10_lowest_growth_year = dt_cresc_pop.groupby('Country/Region')[['Ø Growth/year']].min().reset_index().sort_values('Ø Growth/year')
top10_lowest_growth_year = top10_lowest_growth_year.head(10)

In [178]:
#Plotagem Gráfica
top10_lowest_graphic = px.bar(top10_lowest_growth_year,
                x='Country/Region',
                y='Ø Growth/year',
                title='Os 10 Países com a menor taxa de Crescimento Populacional Anual',
                color='Ø Growth/year',
                color_continuous_scale='blues'
                )

top10_lowest_graphic.update_layout(xaxis_title='País/Região',
                            yaxis_title='Taxa de Crescimento Anual'
                            )

top10_lowest_graphic.show()

Podemos observar que as **Ilhas Marshall** possuem a menor taxa de crescimento anual do mundo! Sendo **-2,98% ao ano.**

#**Taxa de Crescimento Populacional entre 2013 e 2022**

In [179]:
#Realizando a conversão da coluna Growth 2013-2022 para float.
dt_cresc_pop['Growth 2013-2022'] = dt_cresc_pop['Growth 2013-2022'].str.replace('%', '').astype(float)

##**Top 10 Países com o maior Crescimento Populacional entre 2013 e 2022**

In [180]:
top10_growth_decade = dt_cresc_pop.groupby('Country/Region')[['Growth 2013-2022']].max().reset_index().sort_values('Growth 2013-2022', ascending=False)
top10_growth_decade = top10_growth_decade.head(10)

In [181]:
#Plotagem Gráfica
top10_growthDecade_graphic = px.bar(top10_growth_decade,
                x='Country/Region',
                y='Growth 2013-2022',
                title='Os 10 Países com a maior taxa de Crescimento Populacional (2013-2022)',
                color='Growth 2013-2022',
                color_continuous_scale='greens'
                )

top10_growthDecade_graphic.update_layout(xaxis_title='País/Região',
                            yaxis_title='Taxa de Crescimento (2013-2022)'
                            )

top10_growthDecade_graphic.show()

A Guinea Equatorial possui a maior taxa de crescimento entre os anos de 2013 e 2022 com um crescimento de 55.6%.

##**Top 10 Países com o menor Crescimento Populacional entre 2013 e 2022**

In [182]:
top10_lowestGrowth_decade = dt_cresc_pop.groupby('Country/Region')[['Growth 2013-2022']].min().reset_index().sort_values('Growth 2013-2022')
top10_lowestGrowth_decade = top10_lowestGrowth_decade.head(10)

In [183]:
#Plotagem Gráfica
top10_LgrowthDecade_graphic = px.bar(top10_lowestGrowth_decade,
                x='Country/Region',
                y='Growth 2013-2022',
                title='Os 10 Países com a menor taxa de Crescimento Populacional (2013-2022)',
                color='Growth 2013-2022',
                color_continuous_scale='greens'
                )

top10_LgrowthDecade_graphic.update_layout(xaxis_title='País/Região',
                            yaxis_title='Taxa de Crescimento (2013-2022)'
                            )

top10_LgrowthDecade_graphic.show()

As Ilhas Marshall possuem a menor taxa de crescimento entre os anos de 2013 e 2022, com uma **redução de 27%.**

#**Resumo Estatístico**

###Média da Taxa Anual de Crescimento de todos os Países/Regiões

In [184]:
mean_anual_growth = dt_cresc_pop['Ø Growth/year'].mean()
print(f'A média da taxa anual de crescimento populacional de todos os Países e Regiões é: {mean_anual_growth:.4f}')

A média da taxa anual de crescimento populacional de todos os Países e Regiões é: 1.0961


In [185]:
graphic_mean_anual = px.histogram(dt_cresc_pop,
                                  x='Ø Growth/year',
                                  )

graphic_mean_anual.update_layout(title='Distribuição da Média Anual da Taxa de Crescimento Populacional',
                                 xaxis_title='Média da Taxa de Crescimento Populacional Anual',
                                 yaxis_title='Frequência',
                                 )

graphic_mean_anual.show()

###Mediana da Taxa Anual de Crescimento de todos os Países/Regiões

In [186]:
median_anual_growth = dt_cresc_pop['Ø Growth/year'].median()
print(f'A mediana da taxa anual de crescimento populacional de todos os Países e Regiões é: {median_anual_growth}')

A mediana da taxa anual de crescimento populacional de todos os Países e Regiões é: 0.93


In [187]:
dt_boxplot = px.box(dt_cresc_pop,
       y='Ø Growth/year',
       )
dt_boxplot.update_layout(yaxis_title='Taxa de Crescimento Anual')

dt_boxplot.show()

###Média do Crescimento Populacional entre os anos de 2013 e 2022 (Em %)

In [188]:
mean_growth = dt_cresc_pop['Growth 2013-2022'].mean()
print(f'A média do crescimento populacional entre os anos de 2013 e 2022 é de: {mean_growth:.4f}')

A média do crescimento populacional entre os anos de 2013 e 2022 é de: 10.9803


###Mediana do Crescimento Populacional entre os anos de 2013 e 2022 (Em %)

In [189]:
median_growth = dt_cresc_pop['Growth 2013-2022'].median()
print(f'A mediana do crescimento populacional entre os anos de 2013 e 2022 é de: {median_growth}')

A mediana do crescimento populacional entre os anos de 2013 e 2022 é de: 8.41


###Plotagem Scatter da taxa média de crescimento populacional anual x crescimento populacional entre 2013 e 2022

In [190]:
graphic_scatter = px.scatter(dt_cresc_pop, x='Ø Growth/year',
                             y='Growth 2013-2022',
                             title='Scatter da taxa média de crescimento populacional anual x crescimento populacional entre 2013 e 2022',
                             )

graphic_scatter.update_traces(marker=dict(size=8, color='forestgreen', symbol='circle'), selector=dict(mode='markers'))

graphic_scatter.update_layout(
    xaxis_title='Taxa Média do Crescimento Populacional Anual',
    yaxis_title='Crescimento Populacional 2013-2022',
    showlegend=True
)


graphic_scatter.show()